kv/kv.go

```diff
@@ -467,48 +467,37 @@ type Driver interface {
 	Open(path string) (Storage, error)
 }
 
-// TransactionOptionType indicates the method when beginning a transaction
-type TransactionOptionType uint8
-
-const (
-	// WithStartTS indicates the startTS option
-	WithStartTS TransactionOptionType = iota
-	// WithPrevSec indicates the prevSec option
-	WithPrevSec
-)
-
 // TransactionOption indicates the option when beginning a transaction
 type TransactionOption struct {
-	TransactionOptionType
-	StartTS uint64
-	PrevSec uint64
+	TxnScope string
+	StartTS  *uint64
+	PrevSec  *uint64
```

Comment:
I don't think it is a good idea to use pointers for integers in go. It may bring another two escaped memory. Copying a slice of string or integers is free.